basics_of_sst-core/tutorial/Example01.md

```diff
@@ -0,0 +1,436 @@
+```
+
+Then add the link definition following the `setup()` and `finish()` method definitions.
+
+```
+            // Standard SST::Component functions.  These all need to
+            // be implemented in the component, even if they are empty.
+            //
+            void setup(void);
+            void finish(void);
+
+            // Links to connect to other component.
+            // The link connects to the components port.
+            //
+            SST::Link* link_;
+```
+
+Finally, use the `SST_ELI_DOCUMENT_PORTS` macro to define the link in the ELI database.  Notice, this is one of the places where the SST code interface refers to a link as a port.  Place this macro below the previously defined macros.
+
+```
+            // Shared documentation macros. 
+            //
+            SST_ELI_DOCUMENT_PARAMS(
+                { "debug", "Debug location:  (0: NONE, 1: STDOUT, 2: STDERR, 3: FILE)", "0" },
+                { "clock", "Component clock rate", "1GHz" },
+                { "clockTicks", "Number of times the handler is called before ending.", "10" }
+            )
+            
+            SST_ELI_REGISTER_COMPONENT(
+                ExampleComponent,                       // Class name
+                "example",                              // Library name (the *.so)
+                "ExampleComponent",                     // Named use to reference the component.  This can be
+                                                        // whatever you want it to be and will be referenced
+                                                        // in the python configuration file.
+                SST_ELI_ELEMENT_VERSION( 1, 0, 0 ),     // Version number
+                "Clock with port element example",      // Description                 
+                COMPONENT_CATEGORY_UNCATEGORIZED        // Component category
+            )
+
+            SST_ELI_DOCUMENT_PORTS(
+                { "link_", "Single port", {}}
+            )
+```
+
+In the `SSE_ELI_DOCUMENT_PORTS` macro definition the first parameter \("link_"\) is the name assigned to the documented port.  It does not have to match the variable name, this has just been done that to make the connection between the two explicit.  This name will be used later to reference the link in the simulation configuration file.  
+
+The second parameter is a description and can be anything.  The third parameter is not used for this example.
+
+Save this file and open the `ExampleComponent.cc` file where the component functionality is defined.  
+
+In the component constructor, add the code to configure the link above that portion where the component clock is configured
+
+```
+    // Configure the links (connections to other components).
+    // The link is associated with a component's registered port.
+    // No handler for this example.
+    //
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Configuring link.\n");
+    link_ = configureLink("link_");
```

Comment:
It's a good practice to check that link_ != nullptr after calling configureLink() as this would indicate that the user failed to connect the port (or misspelled something, etc.).